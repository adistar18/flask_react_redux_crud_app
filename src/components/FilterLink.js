import React, { Component, PropTypes } from 'react';

export default class Link extends Component {
  render() {
    const { linkText, count, handleClick } = this.props;

    return (
      <a 
        className="list-group-item"
        onClick={handleClick}>
        {linkText}
        {count ? <span className="badge">{count}</span> : null}
      </a>
    );
  }
}

Link.propTypes = {
  handleClick: PropTypes.func.isRequired,
  linkText: PropTypes.string.isRequired,
  count: PropTypes.number
};
