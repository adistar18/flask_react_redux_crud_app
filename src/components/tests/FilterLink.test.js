import React from 'react';
import FilterLink from 'components/FilterLink';
import { shallow } from 'enzyme';

describe('FilterLink', () => {
  let renderedComponent;
  const linkText = "Family Law";
  const handleClick = jest.fn();

  beforeEach(() => {
    renderedComponent = shallow(
      <FilterLink 
        linkText={linkText}
        handleClick={handleClick}/>
    );
  })

  it('should render <a> only if count prop not supplied', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render <a> and <span> if count prop is supplied', () => {
    renderedComponent = shallow(
      <FilterLink 
        linkText={linkText}
        handleClick={handleClick}
        count={2}/>
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should call handleClick when clicked', () => {
    renderedComponent.find('a').simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
})
