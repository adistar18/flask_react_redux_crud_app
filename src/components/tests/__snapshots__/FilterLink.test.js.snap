exports[`FilterLink should render <a> and <span> if count prop is supplied 1`] = `
<a
  className="list-group-item"
  onClick={[Function]}>
  Family Law
  <span
    className="badge">
    2
  </span>
</a>
`;

exports[`FilterLink should render <a> only if count prop not supplied 1`] = `
<a
  className="list-group-item"
  onClick={[Function]}>
  Family Law
</a>
`;
